packages/swapper/src/manager/SwapperManager.test.ts

```diff
@@ -227,11 +228,13 @@ describe('SwapperManager', () => {
 
       const cowSwapperGetUsdTradeQuoteMock = jest
         .spyOn(cowSwapper, 'getTradeQuote')
-        .mockImplementation(jest.fn().mockResolvedValueOnce(badTradeQuote))
+        // TODO(gomes): is mocking OK(badTradeQuote), in other words what's the intent of this?
+        // This previously mocked resolution with a bad trade quote which seems wrong - if there's a "bad" trade quote, means we should bail and not return it?
+        .mockImplementation(jest.fn().mockResolvedValueOnce(Ok(badTradeQuote)))
```

Comment:
If this bad quote is further checked for being bad (has conditional logic applied to it), then we could prob handle it as a type better now. If it's just returned to the client as is then I'd keep it

Reply:
Will have to give this some more testing at runtime, still unsure what bad really meant here 😢. Until I figure it out, keeping the flow the same as is in those tests i.e resolving to a "bad trade quote"